core/cache.go

```diff
@@ -28,23 +28,19 @@ func (*CacheVolume) TypeDescription() string {
 	return "A directory whose contents persist across runs."
 }
 
-func NewCache(keys ...string) *CacheVolume {
-	return &CacheVolume{Keys: keys}
+func NewCache(key string) *CacheVolume {
+	return &CacheVolume{Key: key}
 }
 
 func (cache *CacheVolume) Clone() *CacheVolume {
 	cp := *cache
-	cp.Keys = cloneSlice(cp.Keys)
 	return &cp
 }
 
 // Sum returns a checksum of the cache tokens suitable for use as a cache key.
 func (cache *CacheVolume) Sum() string {
 	hash := sha256.New()
-	for _, tok := range cache.Keys {
-		_, _ = hash.Write([]byte(tok + "\x00"))
-	}
-
+	_, _ = hash.Write([]byte(cache.Key + "\x00"))
```

Comment:
extreme nit, just if you're updating other stuff before merge: the `\x00` isn't needed since it was just a delimiter used before to separate keys. Can remove it now for clarity.